source/Cosmos.System2/Graphics/FullScreenCanvas.cs

```diff
@@ -1,77 +1,99 @@
 //#define COSMOSDEBUG
 using Cosmos.HAL;
+using Cosmos.HAL.Drivers;
 
 namespace Cosmos.System.Graphics
 {
     public static class FullScreenCanvas
     {
-        private static PCIDevice SVGAIIDevice = PCI.GetDevice(VendorID.VMWare, DeviceID.SVGAIIAdapter);
+        public static bool IsInUse = false;
 
-        public static bool SVGAIIExist()
+        public static void Disable()
         {
-            if (SVGAIIDevice == null)
+            if (IsInUse)
             {
-                return false;
+                _VideoDriver.Disable();
+                VGAScreen.SetTextMode(VGADriver.TextSize.Size80x25);
             }
+        }
+
+        private enum VideoDriver
+        {
+            VMWareSVGAIIDriver,
+            VBEDriver,
+            VGADriver
+        }
+
+        private static PCIDevice _SVGAIIDevice = PCI.GetDevice(VendorID.VMWare, DeviceID.SVGAIIAdapter);
 
-            return SVGAIIDevice.DeviceExists;
+        public static bool BGAExists()
+        {
+            return VBEDriver.Available();
         }
 
-        private static Canvas MyVideoDriver = null;
+        private static Canvas _VideoDriver = null;
 
         private static Canvas GetVideoDriver()
         {
-            if (SVGAIIExist())
+            if (_SVGAIIDevice != null && PCI.Exists(_SVGAIIDevice))
+            {
+                return new SVGAIICanvas();
+            }
+            else if (BGAExists() || PCI.GetDevice((VendorID)0x80EE, (DeviceID)0xBEEF) != null)
             {
-                return new SVGAIIScreen();
+                return new VBECanvas();
             }
             else
             {
-                return new VBEScreen();
+                return new VGACanvas();
             }
         }
 
         private static Canvas GetVideoDriver(Mode mode)
         {
-            if (SVGAIIExist())
+            if (_SVGAIIDevice != null && PCI.Exists(_SVGAIIDevice))
+            {
+                return new SVGAIICanvas(mode);
+            }
+            else if (BGAExists() || PCI.GetDevice((VendorID)0x80EE, (DeviceID)0xBEEF) != null)
```

Comment:
VBE is detected for bochs and VirtualBox now, @quajak do you think it has to be detected through Multiboot too? Because VBE VESA can be set by syslinux and the VBE informations are set into multiboot